### ISSUE TITLE
How to test own data

### ISSUE THREAD

**Issue by Flyooofly** [ShiqiYu/OpenGait#61]
## System information (version)

- Pytorch => :grey_question:
- Operating System / Platform => :grey_question:
- Cuda => :grey_question:

## Detailed description
Hi, I would like to use the provided pretrained model for testing. My idea is to prepare a few pictures of the gait that I already know who it is, and then prepare a test picture, and calculate the similarity between them to determine which known given picture the gait of the test picture belongs to. Is there any idea if I want to achieve this?

## Issue submission checklist
 - [ ] I checked the problem with documentation, FAQ, issues, and have not found solution.

**Comment by noahshen98** [comment]
Hi, you can test your own dataset for sure. Such one approach can be labeling your own gallery (known identity) as '**YOUR OWN DATASET**' : '01' like 
https://github.com/ShiqiYu/OpenGait/blob/1be7333c8656223af8eab9c2bc81fd429dfd2f9c/opengait/utils/evaluation.py#L125, and placing your probe (sequence to be tested) as '**YOUR OWN DATASET**'' : '02' .

if you organize your dataset like that, you will obtain a rank list after you run the evaluation.https://github.com/ShiqiYu/OpenGait/blob/1be7333c8656223af8eab9c2bc81fd429dfd2f9c/opengait/utils/evaluation.py#L141

**Comment by Flyooofly** [comment]
ok，thx.



---Original---
From: ***@***.***&gt;
Date: Mon, Jun 13, 2022 19:16 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [ShiqiYu/OpenGait] How to test own data (Issue #61)




 
Hi, you can test your own dataset for sure. Such one approach can be labeling your own gallery (known identity) as 'YOUR OWN DATASET' : '01' like
 https://github.com/ShiqiYu/OpenGait/blob/1be7333c8656223af8eab9c2bc81fd429dfd2f9c/opengait/utils/evaluation.py#L125, and placing your probe (sequence to be tested) as 'YOUR OWN DATASET'' : '02' .
 
if you organize your dataset like that, you will obtain a rank list after you run the evaluation.https://github.com/ShiqiYu/OpenGait/blob/1be7333c8656223af8eab9c2bc81fd429dfd2f9c/opengait/utils/evaluation.py#L141

**Comment by puyiwen** [comment]
> GREW dataset would be a better example, the probe within the Test set doesn't have its corresponding ground truth, and the output is a ranking list for each probe sequence.
> 
> https://github.com/ShiqiYu/OpenGait/blob/1be7333c8656223af8eab9c2bc81fd429dfd2f9c/opengait/utils/evaluation.py#L179

Hi, what do you mean " rank1, rank2....rank20"?
